pkg/osvscanner/osvscanner.go

```diff
@@ -862,7 +862,7 @@ func filterUnscannablePackages(packages []scannedPackage) []scannedPackage {
 
 // patchPackageForRequest modifies packages before they are sent to osv.dev to
 // account for edge cases.
-func patchPackageForRequest(pkg scannedPackage) scannedPackage {
+func patchPackageForRequest(pkg *scannedPackage) *scannedPackage {
```

Comment:
This should not be a pointer I believe. The passed in argument should be copied/immutable, the returned value is the patched version.